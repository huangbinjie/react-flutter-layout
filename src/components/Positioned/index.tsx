import React, { PropsWithChildren } from 'react'
import { Container } from '../..';

export type PositionedProps = {
  top: string
  left: string
  bottom: string
  right: string
}

export default function Positioned(props: PropsWithChildren<PositionedProps>) {
  const {
    children,
    top,
    right,
    left,
    bottom
  } = props

  return <Container debuggerLabel="Positioned" style={{ position: 'absolute', top, bottom, left, right }}>{children}</Container>
}
