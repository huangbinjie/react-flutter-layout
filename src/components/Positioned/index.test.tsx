import React from 'react'
import renderer from 'react-test-renderer'

import Positioned from '.'

describe('Positioned', () => {
  test('basic', () => {
    const tree = renderer.create(
      <Positioned>empty content</Positioned>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
