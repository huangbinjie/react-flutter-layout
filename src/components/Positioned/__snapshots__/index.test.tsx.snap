// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Positioned basic 1`] = `
<div
  className="fhwmucx"
  debug-label="Positioned"
>
  empty content
</div>
`;
